xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -318,10 +318,12 @@ $xwiki.getUserName("xwiki:${username}")
     ## We avoid the whitespace because the users are displayed as inline blocks.
     #set ($discard = $output.addAll([
       "<$userTag class=""$type"" data-reference=""$escapetool.xml($userReference)"">",
+        "<$userNameTag $userNameTagAttributes>",
```

Comment:
This change feels a bit dangerous: you're changing the DOM for the display of `#displayUser` which is a very used macro. It's probably harmless but at the very least I would run a few integration tests, maybe flamingo-skin-test-docker. 

And we probably should discuss such change in the future.

Reply:
Couldn't you match the `user-name` or `group-name` class attribute instead of modifying the HTML?

Reply:
> And we probably should discuss such change in the future.

Created https://forum.xwiki.org/t/proposal-update-the-structure-of-the-user-display-macro/14723 :+1:

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/3085/commits/15a5e5f4f759fa6c4ffbbc39efa1a958fc2da393 :+1: